shared/dataflow/codeql/dataflow/DataFlow.qll

```diff
@@ -30,9 +33,27 @@ signature module InputSig {
     ReturnKind getKind();
   }
 
+  /**
+   * A node in the data flow graph that represents an output.
+   */
   class OutNode extends Node;
 
+  /**
+   * A node in the data flow graph representing the value of an argument after
+   * an operation that might have changed its state. For example, consider the
+   * following C++ code:
+   * ```
+   * int a = 1;
+   * increment(&a);
+   * ```
+   * The post-update node for `&a` represents the value of `&a` after
+   * modification by the call to `increment`.
```

Comment:
This is inaccurate, as it's not just arguments. Also, I'm not completely sure about the example, as I don't know the exact details of the C++ modelling in this case - IIRC the value of a dereference in C++ isn't currently handled by field flow, so the example certainly isn't canonical.
```suggestion
   * A node in the data flow graph representing the value of some other node
   * after an operation that might have changed its state. A typical example is
   * an argument, which may have been modified by the callee. For example,
   * consider the following code calling a setter method:
   * ```
   * x.setFoo(y);
   * ```
   * The post-update node for the argument node `x` is the node representing the
   * value of `x` after the field `foo` has been updated.
```

Reply:
This is really helpful, thanks!